README.md
Gravatar
======
Gravatar jest wtyczką do komunikatora [AQQ](http://www.aqq.eu/pl.php). Pilnuje, aby we wszystkich sieciach był ustawiony zawsze aktualny awatar określony w serwisie gravatar.com.

Błędy
-------
Znalezione błędy wtyczki należy zgłaszać na [tracekerze](http://forum.aqq.eu/tracker/project-81-gravatar/) znajdującym na oficjalnym forum komunikatora AQQ lub pisząc bezpośrednio do autora wtyczki (preferowany kontakt poprzez Jabber).

Kontakt z autorem
-------
Autorem wtyczki Gravatar jest Krzysztof Grochocki. Możesz skontaktować się z nim drogą mailową (kontakt@beherit.pl) lub poprzez Jabber (im@beherit.pl).

Licencja
-------
Wtyczka Gravatar objęta jest licencją [GNU General Public License 3](http://www.gnu.org/copyleft/gpl.html).

    Gravatar
    Copyright (C) 2010-2014  Krzysztof Grochocki

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.

CHANGELOG.md
1.4.0.0
-----
* Kompilacja pod poprawionymi nagłówkami dla wersji 64-bitowej.
* Zmniejszenie rozmiaru wtyczki 64-bitowej poprzez użycie kompresji LZMA.
* Nieskórkowanie systemowych okien dialogowych niepochodzących z wtyczki.
* Aktualizacja komponentów AlphaControls do nowszej wersji (9.10).

1.3.1.0
-----
* Obsługa kolorystyki stylizacji okien (barwa i nasycenie).
* Drobna poprawka w obsłudze nadawania przeźroczystości obramowań przez wtyczkę AlphaWindows.
* Aktualizacja komponentów AlphaControls do nowszej wersji (8.44).

1.3.0.0
-----
* Dodanie 64-bitowej wersji wtyczki.
* Wsparcie dla wtyczki AlphaWindows (ustawianie przeźroczystości obramowania).
* Drobne zmiany w skórkowaniu form ustawień.
* Zmiana sposobu wypakowywania plików z zasobów wtyczki.
* Zmniejszenie rozmiaru wtyczki 32-bitowej poprzez użycie kompresji LZMA.
* Aktualizacja komponentów AlphaControls do nowszej wersji (8.20).

1.2.0.0
-----
* Przepisanie wtyczki pod kompilator Embarcadero C++ Builder XE3.
* Aktualizacja komponentów AlphaControls do najnowszej dostępnej wersji (7.70).
* Inne mniej ważne zmiany w kodzie.

1.1.3.0
-----
* Wtyczka posiada własną ikonę pokazywaną w ustawieniach komunikatora na liście wtyczek.
* Przy aktualizacji awatara we wtyczce SkypeCore pokazywany jest tylko identyfikator Skype.
* Atualizacja komponentów AlphaControls do najnowszej dostępnej wersji (7.65).
* Obsługa natywnego stylu Windows (wyłączone skórkowanie okien wtyczki).
* Optymalizacja kodu.

1.1.2.0
-----
* Opcja wymuszania aktualizacji.
* Poprawki w oprawie tekstowej form wtyczki.

1.1.0.0
-----
* Pełne skórkowanie wszystkich okien wtyczki.
* Aktualizowanie awatarów we wtyczce GG Network.
* Komunikat z potwierdzeniem dokonania aktualizacji po zmianach w ustawieniach wtyczki.
* Poprawki kosmetyczne formy pierwszego uruchomienia wtyczki.
* Zmniejszono czas wyświetlania chmurek informacyjnych.
* Zmiana sposobu pobierania awatarów oraz rozszerzenie informowania o niepomyślnej aktualizacji.
* Spora optymalizacja kodu.

1.0.0.0
-----
* Pierwsza publiczna wersja.

Gravatar.h
//---------------------------------------------------------------------------
// Copyright (C) 2010-2014 Krzysztof Grochocki
//
// This file is part of Gravatar
//
// Gravatar is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// Gravatar is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with GNU Radio; see the file COPYING. If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.
//---------------------------------------------------------------------------

#include <vcl.h>
#include <tchar.h>


FirstRunFrm.h
//---------------------------------------------------------------------------
// Copyright (C) 2010-2013 Krzysztof Grochocki
//
// This file is part of Gravatar
//
// Gravatar is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// Gravatar is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with GNU Radio; see the file COPYING. If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.
//---------------------------------------------------------------------------

//---------------------------------------------------------------------------
#ifndef FirstRunFrmH
#define FirstRunFrmH
#define WM_ALPHAWINDOWS (WM_USER + 666)
//---------------------------------------------------------------------------
#include "acPNG.hpp"
#include "sBevel.hpp"
#include "sButton.hpp"
#include "sCheckBox.hpp"
#include "sCheckListBox.hpp"
#include "sComboBox.hpp"
#include "sEdit.hpp"
#include "sLabel.hpp"
#include "sListBox.hpp"
#include "sPanel.hpp"
#include "sRadioButton.hpp"
#include "sSkinManager.hpp"
#include "sSkinProvider.hpp"
#include <System.Actions.hpp>
#include <System.Classes.hpp>
#include <Vcl.ActnList.hpp>
#include <Vcl.Controls.hpp>
#include <Vcl.ExtCtrls.hpp>
#include <Vcl.Imaging.jpeg.hpp>
#include <Vcl.StdCtrls.hpp>
//---------------------------------------------------------------------------
class TFirstRunForm : public TForm
{
__published:	// IDE-managed Components
	TsBevel *Bevel;
	TsPanel *Panel1;
	TsButton *NextButton;
	TsButton *PrevButton;
	TsButton *SaveButton;
	TActionList *ActionList;
	TAction *aSaveSettings;
	TsLabel *WelcomeLabel1;
	TImage *GravatarImage;
	TsLabel *WelcomeLabel2;
	TsLabel *WelcomeLabel3;
	TsPanel *Panel2;
	TsLabel *GetModeLabel;
	TsRadioButton *GetMode0RadioButton;
	TsRadioButton *GetMode1RadioButton;
	TsEdit *EmailEdit;
	TsPanel *Panel3;
	TsLabel *DefaultAvatarLabel;
	TsRadioButton *DefaultAvatarRadioButton0;
	TsRadioButton *DefaultAvatarRadioButton1;
	TsRadioButton *DefaultAvatarRadioButton2;
	TsRadioButton *DefaultAvatarRadioButton3;
	TsPanel *Panel4;
	TsLabel *IntervalLabel;
	TsComboBox *IntervalComboBox;
	TsCheckBox *InfoSuccessCheckBox;
	TsPanel *Panel5;
	TsLabel *InfoLabel;
	TsCheckBox *InfoFailCheckBox;
	TsLabel *AccountsLabel;
	TsRadioButton *AccountsMode0RadioButton;
	TsRadioButton *AccountsMode1RadioButton;
	TsCheckListBox *AccountsCheckListBox;
	TsPanel *Panel6;
	TsLabel *EndLabel1;
	TsLabel *EndLabel2;
	TImage *GravatarImage2;
	TsLabel *EndLabel3;
	TsSkinManager *sSkinManager;
	TsSkinProvider *sSkinProvider;
	TAction *aNextButton;
	TAction *aPrevButton;
	TAction *aDefaultAvatar0;
	TAction *aDefaultAvatar1;
	TAction *aDefaultAvatar2;
	TAction *aDefaultAvatar3;
	TsCheckBox *ForceUpdateCheckBox;
	TsBevel *DefaultAvatarBevel0;
	TsBevel *DefaultAvatarBevel1;
	TsBevel *DefaultAvatarBevel2;
	TsBevel *DefaultAvatarBevel3;
	TImage *DefaultAvatarImage0;
	TImage *DefaultAvatarImage1;
	TImage *DefaultAvatarImage2;
	TImage *DefaultAvatarImage3;
	void __fastcall SaveButtonClick(TObject *Sender);
	void __fastcall GetMode1RadioButtonClick(TObject *Sender);
	void __fastcall GetMode0RadioButtonClick(TObject *Sender);
	void __fastcall DefaultAvatarImage0Click(TObject *Sender);
	void __fastcall DefaultAvatarImage1Click(TObject *Sender);
	void __fastcall DefaultAvatarImage2Click(TObject *Sender);
	void __fastcall DefaultAvatarImage3Click(TObject *Sender);
	void __fastcall FormShow(TObject *Sender);
	void __fastcall AccountsMode1RadioButtonClick(TObject *Sender);
	void __fastcall AccountsMode0RadioButtonClick(TObject *Sender);
	void __fastcall aSaveSettingsExecute(TObject *Sender);
	void __fastcall FormCreate(TObject *Sender);
	void __fastcall aNextButtonExecute(TObject *Sender);
	void __fastcall aPrevButtonExecute(TObject *Sender);
	void __fastcall aDefaultAvatar0Execute(TObject *Sender);
	void __fastcall aDefaultAvatar1Execute(TObject *Sender);
	void __fastcall aDefaultAvatar2Execute(TObject *Sender);
	void __fastcall aDefaultAvatar3Execute(TObject *Sender);
	void __fastcall ForceUpdateCheckBoxClick(TObject *Sender);
	void __fastcall sSkinManagerSysDlgInit(TacSysDlgData DlgData, bool &AllowSkinning);
private:	// User declarations
public:		// User declarations
	__fastcall TFirstRunForm(TComponent* Owner);
	void __fastcall WMTransparency(TMessage &Message);
	BEGIN_MESSAGE_MAP
	MESSAGE_HANDLER(WM_ALPHAWINDOWS,TMessage,WMTransparency);
	END_MESSAGE_MAP(TForm)
};
//---------------------------------------------------------------------------
extern PACKAGE TFirstRunForm *FirstRunForm;
//---------------------------------------------------------------------------
#endif


GravatarFrm.h
//---------------------------------------------------------------------------
// Copyright (C) 2010-2013 Krzysztof Grochocki
//
// This file is part of Gravatar
//
// Gravatar is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// Gravatar is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with GNU Radio; see the file COPYING. If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.
//---------------------------------------------------------------------------

//---------------------------------------------------------------------------
#ifndef GravatarFrmH
#define GravatarFrmH
#define WM_ALPHAWINDOWS (WM_USER + 666)
//---------------------------------------------------------------------------
#include "acPNG.hpp"
#include "sBevel.hpp"
#include "sButton.hpp"
#include "sCheckBox.hpp"
#include "sCheckListBox.hpp"
#include "sComboBox.hpp"
#include "sEdit.hpp"
#include "sLabel.hpp"
#include "sListBox.hpp"
#include "sPageControl.hpp"
#include "sRadioButton.hpp"
#include "sSkinManager.hpp"
#include "sSkinProvider.hpp"
#include <IdAntiFreezeBase.hpp>
#include <IdBaseComponent.hpp>
#include <IdComponent.hpp>
#include <IdHTTP.hpp>
#include <IdTCPClient.hpp>
#include <IdTCPConnection.hpp>
#include <System.Actions.hpp>
#include <System.Classes.hpp>
#include <Vcl.ActnList.hpp>
#include <Vcl.ComCtrls.hpp>
#include <Vcl.Controls.hpp>
#include <Vcl.ExtCtrls.hpp>
#include <Vcl.IdAntiFreeze.hpp>
#include <Vcl.Imaging.jpeg.hpp>
#include <Vcl.StdCtrls.hpp>
//---------------------------------------------------------------------------
class TGravatarForm : public TForm
{
__published:	// IDE-managed Components
	TIdHTTP *IdHTTP;
	TsPageControl *PageControl;
	TsTabSheet *DefaultAvatarTabSheet;
	TsButton *SaveButton;
	TsBevel *Bevel;
	TActionList *ActionList;
	TAction *aExit;
	TAction *aLoadSettings;
	TAction *aSaveSettings;
	TsTabSheet *EmailTabSheet;
	TsEdit *EmailEdit;
	TsRadioButton *GetMode0RadioButton;
	TsRadioButton *GetMode1RadioButton;
	TsLabel *GetModeLabel;
	TsBevel *DefaultAvatarBevel0;
	TsBevel *DefaultAvatarBevel1;
	TsBevel *DefaultAvatarBevel2;
	TsBevel *DefaultAvatarBevel3;
	TImage *DefaultAvatarImage0;
	TImage *DefaultAvatarImage1;
	TImage *DefaultAvatarImage2;
	TImage *DefaultAvatarImage3;
	TsLabel *DefaultAvatarLabel;
	TsRadioButton *DefaultAvatarRadioButton0;
	TsRadioButton *DefaultAvatarRadioButton1;
	TsRadioButton *DefaultAvatarRadioButton2;
	TsRadioButton *DefaultAvatarRadioButton3;
	TsTabSheet *UpdateTabSheet;
	TsCheckBox *InfoFailCheckBox;
	TsLabel *InfoLabel;
	TsCheckBox *InfoSuccessCheckBox;
	TsComboBox *IntervalComboBox;
	TsLabel *IntervalLabel;
	TsTabSheet *AccountsTabSheet;
	TsCheckListBox *AccountsCheckListBox;
	TsLabel *AccountsLabel;
	TsRadioButton *AccountsMode0RadioButton;
	TsRadioButton *AccountsMode1RadioButton;
	TsBevel *LineBevel;
	TTimer *Timer;
	TTimer *StartTimer;
	TsSkinManager *sSkinManager;
	TsSkinProvider *sSkinProvider;
	TsButton *CancelButton;
	TsButton *OKButton;
	TAction *aDefaultAvatar0;
	TAction *aDefaultAvatar1;
	TAction *aDefaultAvatar2;
	TAction *aDefaultAvatar3;
	TAction *aAllowApply;
	TsCheckBox *ForceUpdateCheckBox;
	TIdAntiFreeze *IdAntiFreeze;
	void __fastcall SaveButtonClick(TObject *Sender);
	void __fastcall aExitExecute(TObject *Sender);
	void __fastcall FormShow(TObject *Sender);
	void __fastcall aLoadSettingsExecute(TObject *Sender);
	void __fastcall aSaveSettingsExecute(TObject *Sender);
	void __fastcall DefaultAvatarImage0Click(TObject *Sender);
	void __fastcall DefaultAvatarImage1Click(TObject *Sender);
	void __fastcall DefaultAvatarImage2Click(TObject *Sender);
	void __fastcall DefaultAvatarImage3Click(TObject *Sender);
	void __fastcall GetMode0RadioButtonClick(TObject *Sender);
	void __fastcall GetMode1RadioButtonClick(TObject *Sender);
	void __fastcall AccountsMode0RadioButtonClick(TObject *Sender);
	void __fastcall AccountsMode1RadioButtonClick(TObject *Sender);
	void __fastcall StartTimerTimer(TObject *Sender);
	void __fastcall TimerTimer(TObject *Sender);
	void __fastcall FormCreate(TObject *Sender);
	void __fastcall aDefaultAvatar0Execute(TObject *Sender);
	void __fastcall aDefaultAvatar1Execute(TObject *Sender);
	void __fastcall aDefaultAvatar2Execute(TObject *Sender);
	void __fastcall aDefaultAvatar3Execute(TObject *Sender);
	void __fastcall OKButtonClick(TObject *Sender);
	void __fastcall aAllowApplyExecute(TObject *Sender);
	void __fastcall ForceUpdateCheckBoxClick(TObject *Sender);
	void __fastcall sSkinManagerSysDlgInit(TacSysDlgData DlgData, bool &AllowSkinning);
private:	// User declarations
	void __fastcall DeleteFiles(UnicodeString DirName, UnicodeString Prefix);
public:		// User declarations
	__fastcall TGravatarForm(TComponent* Owner);
	void __fastcall WMTransparency(TMessage &Message);
	BEGIN_MESSAGE_MAP
	MESSAGE_HANDLER(WM_ALPHAWINDOWS,TMessage,WMTransparency);
	END_MESSAGE_MAP(TForm)
};
//---------------------------------------------------------------------------
extern PACKAGE TGravatarForm *GravatarForm;
//---------------------------------------------------------------------------
#endif


FirstRunFrm.cpp
//---------------------------------------------------------------------------
// Copyright (C) 2010-2014 Krzysztof Grochocki
//
// This file is part of Gravatar
//
// Gravatar is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// Gravatar is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with GNU Radio; see the file COPYING. If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.
//---------------------------------------------------------------------------

//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop
#include "FirstRunFrm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "acPNG"
#pragma link "sBevel"
#pragma link "sButton"
#pragma link "sCheckBox"
#pragma link "sCheckListBox"
#pragma link "sComboBox"
#pragma link "sEdit"
#pragma link "sLabel"
#pragma link "sListBox"
#pragma link "sPanel"
#pragma link "sRadioButton"
#pragma link "sSkinManager"
#pragma link "sSkinProvider"
#pragma resource "*.dfm"
TFirstRunForm *FirstRunForm;
//---------------------------------------------------------------------------
__declspec(dllimport)UnicodeString GetPluginUserDir();
__declspec(dllimport)UnicodeString GetThemeSkinDir();
__declspec(dllimport)bool ChkSkinEnabled();
__declspec(dllimport)bool ChkThemeAnimateWindows();
__declspec(dllimport)bool ChkThemeGlowing();
__declspec(dllimport)int GetSaturation();
__declspec(dllimport)int GetHUE();
__declspec(dllimport)void GetAccountList(bool FirstRun);
__declspec(dllimport)void RefreshAvatars();
__declspec(dllimport)void LoadSettings(bool OnLoad);
//---------------------------------------------------------------------------
__fastcall TFirstRunForm::TFirstRunForm(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::WMTransparency(TMessage &Message)
{
  Application->ProcessMessages();
  sSkinProvider->BorderForm->UpdateExBordersPos(true,(int)Message.LParam);
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::FormCreate(TObject *Sender)
{
  //Wlaczona zaawansowana stylizacja okien
  if(ChkSkinEnabled())
  {
	UnicodeString ThemeSkinDir = GetThemeSkinDir();
	//Plik zaawansowanej stylizacji okien istnieje
	if(FileExists(ThemeSkinDir + "\\\\Skin.asz"))
	{
	  //Dane pliku zaawansowanej stylizacji okien
	  ThemeSkinDir = StringReplace(ThemeSkinDir, "\\\\", "\\", TReplaceFlags() << rfReplaceAll);
	  sSkinManager->SkinDirectory = ThemeSkinDir;
	  sSkinManager->SkinName = "Skin.asz";
	  //Ustawianie animacji AlphaControls
	  if(ChkThemeAnimateWindows()) sSkinManager->AnimEffects->FormShow->Time = 200;
	  else sSkinManager->AnimEffects->FormShow->Time = 0;
	  sSkinManager->Effects->AllowGlowing = ChkThemeGlowing();
	  //Zmiana kolorystyki AlphaControls
	  sSkinManager->HueOffset = GetHUE();
	  sSkinManager->Saturation = GetSaturation();
	  //Aktywacja skorkowania AlphaControls
	  sSkinManager->Active = true;
	}
	//Brak pliku zaawansowanej stylizacji okien
	else sSkinManager->Active = false;
  }
  //Zaawansowana stylizacja okien wylaczona
  else sSkinManager->Active = false;
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::FormShow(TObject *Sender)
{
  //Pobieranie listy kont
  GetAccountList(true);
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::aSaveSettingsExecute(TObject *Sender)
{
  TIniFile *Ini = new TIniFile(GetPluginUserDir() + "\\\\Gravatar\\\\Settings.ini");
  //Domyslny awatar
  if(DefaultAvatarRadioButton0->Checked)
   Ini->WriteInteger("Settings","DefaultAvatar",0);
  else if(DefaultAvatarRadioButton1->Checked)
   Ini->WriteInteger("Settings","DefaultAvatar",1);
  else if(DefaultAvatarRadioButton2->Checked)
   Ini->WriteInteger("Settings","DefaultAvatar",2);
  else
   Ini->WriteInteger("Settings","DefaultAvatar",3);
  //Sposob pobierania adresu e-mail
  Ini->WriteBool("Settings","GetMode",!GetMode0RadioButton->Checked);
  //Ustalony przez usera adres e-mail
  if(EmailEdit->Text=="Wpisz tutaj adres e-mail") EmailEdit->Text = "";
  Ini->WriteString("Settings","StaticEmail",EmailEdit->Text);
   //Czestotliwosc sprawdzania aktualizacji
  Ini->WriteInteger("Settings","Interval",IntervalComboBox->ItemIndex);
  //Wymuszanie aktualizacji
  Ini->WriteBool("Settings","ForceUpdate",ForceUpdateCheckBox->Checked);
  //Informacja o pomyslnym zaktualizowaniu awatarow
  Ini->WriteBool("Settings","InfoSuccess",InfoSuccessCheckBox->Checked);
  //Informacja o bledzie podczas aktualizacji awatarow
  Ini->WriteBool("Settings","InfoFail",InfoFailCheckBox->Checked);
  //Tryb obslugi kont
  Ini->WriteBool("Settings","AccountsMode",!AccountsMode0RadioButton->Checked);
  //Lista obslugiwanych kont
  UnicodeString pAccounts;
  for(int Count=0;Count<AccountsCheckListBox->Items->Count;Count++)
  {
	if(AccountsCheckListBox->Checked[Count])
	 pAccounts = pAccounts + AccountsCheckListBox->Items->Strings[Count] + ";";
  }
  Ini->WriteString("Settings","Accounts",pAccounts);
  delete Ini;
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::aNextButtonExecute(TObject *Sender)
{
  if(Panel1->Visible)
  {
	Panel1->Visible = false;
	Panel2->Visible = true;
	aPrevButton->Enabled = true;
	NextButton->SetFocus();
  }
  else if(Panel2->Visible)
  {
	if((GetMode0RadioButton->Checked)||((GetMode1RadioButton->Checked)&&(!EmailEdit->Text.IsEmpty())))
	{
	  Panel2->Visible = false;
	  Panel3->Visible = true;
	  NextButton->SetFocus();
	}
  }
  else if(Panel3->Visible)
  {
	Panel3->Visible = false;
	Panel4->Visible = true;
	NextButton->SetFocus();
  }
  else if(Panel4->Visible)
  {
	Panel4->Visible = false;
	Panel5->Visible = true;
	NextButton->SetFocus();
  }
  else if(Panel5->Visible)
  {
	Panel5->Visible = false;
	Panel6->Visible = true;
	aNextButton->Visible = false;
	SaveButton->Visible = true;
	SaveButton->SetFocus();
  }
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::aPrevButtonExecute(TObject *Sender)
{
  if(Panel2->Visible)
  {
	Panel2->Visible = false;
	Panel1->Visible = true;
	aPrevButton->Enabled = false;
	NextButton->SetFocus();
  }
  else if(Panel3->Visible)
  {
	Panel3->Visible = false;
	Panel2->Visible = true;
	PrevButton->SetFocus();
  }
  else if(Panel4->Visible)
  {
	Panel4->Visible = false;
	Panel3->Visible = true;
	PrevButton->SetFocus();
  }
  else if(Panel5->Visible)
  {
	Panel5->Visible = false;
	Panel4->Visible = true;
	PrevButton->SetFocus();
  }
  else if(Panel6->Visible)
  {
	Panel6->Visible = false;
	Panel5->Visible = true;
	aNextButton->Visible = true;
	SaveButton->Visible = false;
	PrevButton->SetFocus();
  }
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::GetMode0RadioButtonClick(TObject *Sender)
{
  if(EmailEdit->Text.IsEmpty()) EmailEdit->Text = "Wpisz tutaj adres e-mail";
  EmailEdit->Enabled = false;
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::GetMode1RadioButtonClick(TObject *Sender)
{
  if(EmailEdit->Text=="Wpisz tutaj adres e-mail") EmailEdit->Text = "";
  EmailEdit->Enabled = true;
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::DefaultAvatarImage0Click(TObject *Sender)
{
  DefaultAvatarRadioButton0->Checked = true;
  aDefaultAvatar0->Execute();
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::DefaultAvatarImage1Click(TObject *Sender)
{
  DefaultAvatarRadioButton1->Checked = true;
  aDefaultAvatar1->Execute();
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::DefaultAvatarImage2Click(TObject *Sender)
{
  DefaultAvatarRadioButton2->Checked = true;
  aDefaultAvatar2->Execute();
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::DefaultAvatarImage3Click(TObject *Sender)
{
  DefaultAvatarRadioButton3->Checked = true;
  aDefaultAvatar3->Execute();
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::AccountsMode0RadioButtonClick(TObject *Sender)
{
  AccountsCheckListBox->Enabled = false;
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::AccountsMode1RadioButtonClick(TObject *Sender)
{
  AccountsCheckListBox->Enabled = true;
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::aDefaultAvatar0Execute(TObject *Sender)
{
  DefaultAvatarBevel0->Visible = true;
  DefaultAvatarBevel1->Visible = false;
  DefaultAvatarBevel2->Visible = false;
  DefaultAvatarBevel3->Visible = false;
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::aDefaultAvatar1Execute(TObject *Sender)
{
  DefaultAvatarBevel0->Visible = false;
  DefaultAvatarBevel1->Visible = true;
  DefaultAvatarBevel2->Visible = false;
  DefaultAvatarBevel3->Visible = false;
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::aDefaultAvatar2Execute(TObject *Sender)
{
  DefaultAvatarBevel0->Visible = false;
  DefaultAvatarBevel1->Visible = false;
  DefaultAvatarBevel2->Visible = true;
  DefaultAvatarBevel3->Visible = false;
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::aDefaultAvatar3Execute(TObject *Sender)
{
  DefaultAvatarBevel0->Visible = false;
  DefaultAvatarBevel1->Visible = false;
  DefaultAvatarBevel2->Visible = false;
  DefaultAvatarBevel3->Visible = true;
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::ForceUpdateCheckBoxClick(TObject *Sender)
{
  InfoSuccessCheckBox->Enabled = !ForceUpdateCheckBox->Checked;
  InfoFailCheckBox->Enabled = !ForceUpdateCheckBox->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::SaveButtonClick(TObject *Sender)
{
  //Zapisanie ustawien
  aSaveSettings->Execute();
  //Odczyt ustawien w rdzeniu wtyczki
  LoadSettings(false);
  //Zamkniecie formy
  Close();
  //Odswiezanie awatarow
  RefreshAvatars();
}
//---------------------------------------------------------------------------

void __fastcall TFirstRunForm::sSkinManagerSysDlgInit(TacSysDlgData DlgData, bool &AllowSkinning)
{
  AllowSkinning = false;
}
//---------------------------------------------------------------------------

